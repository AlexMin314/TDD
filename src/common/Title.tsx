import React from 'react';

export interface ITitleProps {
  title?: string;
  name?: string;
}

const TitleWrapper = ({ children, testId }: any) => (
  <div data-test-id={testId}>
    {children}
  </div>
);

const Title = ({ title, name }: ITitleProps) => (
  <TitleWrapper testId={title}>
    <b>{title}</b>
    <span>{name}</span>
  </TitleWrapper>
)

export default Title;