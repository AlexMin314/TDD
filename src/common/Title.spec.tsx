import React from 'react';
import { mount } from 'enzyme';
import Title from './Title';

// emzyme
describe('<Title />', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<Title title={'test'} name={'alex'} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders title and name', () => {
    const wrapper = mount(<Title title={'test'} name={'alex'} />);
    expect(wrapper.props().title).toBe('test');
    expect(wrapper.props().name).toBe('alex');

    // Dom
    const boldEle = wrapper.find('b');
    expect(boldEle.contains('test')).toBe(true);

    const wrapperComponent = wrapper.find('TitleWrapper');
    expect((wrapperComponent.props() as any).testId).toBe('test');
  });
});