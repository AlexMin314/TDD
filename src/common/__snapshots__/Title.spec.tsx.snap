// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> matches snapshot 1`] = `
<Title
  name="alex"
  title="test"
>
  <TitleWrapper
    testId="test"
  >
    <div
      data-test-id="test"
    >
      <b>
        test
      </b>
      <span>
        alex
      </span>
    </div>
  </TitleWrapper>
</Title>
`;
